### ISSUE TITLE
Using --headerless on --from-csv removes the first row

### ISSUE THREAD

**Issue by jonathandturner** [nushell/nushell#875]
Example:

```
open caco3_plastics.csv --raw | from-csv
━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━
 # │ importe │ shipper │ tariff_ │ name    │ origin  │ shipped │ arrived │ net_wei │ fob_pri │ cif_pri │ ...
   │ r       │         │ item    │         │         │ _at     │ _at     │ ght     │ ce      │ ce      │
───┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────
 0 │ PLASTIC │ S A     │ 2509000 │ CARBONA │ SPAIN   │ 18/03/2 │ 17/04/2 │ 81,000. │ 14,417. │ 18,252. │ ...
   │ OS      │ REVERTE │ 000     │ TO DE   │         │ 016     │ 016     │ 00      │ 58      │ 34      │
   │ RIVAL   │         │         │ CALCIO  │         │         │         │         │         │         │
   │ CIA     │         │         │ TIPO    │         │         │         │         │         │         │
   │ LTDA    │         │         │ CALCIPO │         │         │         │         │         │         │
   │         │         │         │ RE 160  │         │         │         │         │         │         │
   │         │         │         │ T AL    │         │         │         │         │         │         │
```

```
open caco3_plastics.csv --raw | from-csv --headerless
━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━━━━━┯━━━━━
 # │ Column1 │ Column2 │ Column3 │ Column4 │ Column5 │ Column6 │ Column7 │ Column8 │ Column9 │ Column1 │ ...
   │         │         │         │         │         │         │         │         │         │ 0       │
───┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────
 0 │ PLASTIC │ S A     │ 2509000 │ CARBONA │ SPAIN   │ 18/03/2 │ 17/04/2 │ 81,000. │ 14,417. │ 18,252. │ ...
   │ OS      │ REVERTE │ 000     │ TO DE   │         │ 016     │ 016     │ 00      │ 58      │ 34      │
   │ RIVAL   │         │         │ CALCIO  │         │         │         │         │         │         │
   │ CIA     │         │         │ TIPO    │         │         │         │         │         │         │
   │ LTDA    │         │         │ CALCIPO │         │         │         │         │         │         │
   │         │         │         │ RE 160  │         │         │         │         │         │         │
   │         │         │         │ T AL    │         │         │         │         │         │         │
```

I would expect the data in the first row of the csv file to be the first row of the table when using --headerless

**Comment by t-hart** [comment]
This is also the case with `from-ssv` as it was modeled after the `from-csv` command, as briefly mentioned in [this PR](https://github.com/nushell/nushell/pull/830#issue-327987498). I agree that it's counterintuitive and would be interested in making the required changes, especially in the `from-ssv` command, but would also be happy to take on the task in `from-csv` if no-one else wants it.

**Comment by andrasio** [comment]
@t-hart Please do! `from-tsv` as well. We are also interested in renaming the flags for each of these from `--headerless` to `--per-line`, if you are interested in these changes.

Also `from-json` renaming `--objects` to `--per-line`.

**Comment by t-hart** [comment]
@andrasio Aight, cool. I'll get started on that over the next week, then. Seems like it'll be quite a few changes, so I'll stage them up in multiple PRs. If anyone else fancies doing any of them, I'd be happy to share too :) 

Regarding the flag renaming, what's the logic behind the new name? `--headerless` makes it sound as if it's a file without headers to me, but I'm not sure what the `--per-line` flag means. Also feels like the task isn't directly related, so maybe it would be better to create a separate issue for it?

**Comment by t-hart** [comment]
I've been doing some scouting today and checked out the `csv` crate that's in use by `from-csv` and `from-tsv`: it's got a `has_headers` configuration option which is currently always being set to false. Using the headers functionality is probably the easiest way around this (as in [this example](https://docs.rs/csv/1.1.1/csv/struct.Reader.html#method.headers) from the docs). 

I notice that there is a striking similarity between the implementations of the `tsv` and `csv` commands, and was wondering whether they should/could be refactored into a shared set of functions, but I'm not sure whether this is accidental duplication or whether they should always evolve the same. I'm leaning towards the latter, but I'd like some input on that.

Last: for the `from-ssv` command, we discussed in #845 whether to go with a strict delimiter or to use alignment with headers, and we landed on the latter. In the event that there are no headers, this obviously becomes a bit trickier, and there are several ways to go about it: my first instinct is to use a strict delimiter in this case, but we could also parse the rows and try and figure out what the structure is based on the values. That might sound a bit iffy, but I think it could work quite well; and in the event that the input isn't well-structured, we could inform the user. This is similar to how `csv` and `tsv` require each row to have an equal number of columns, lest they error saying the input cannot be parsed as `(c/t)sv`.

**Comment by andrasio** [comment]
@t-hart Agreed. In that case let's first start working on the equivalent for `--headerless` first and revisit `--per-line` in the future.

**Comment by coolshaurya** [comment]
shouldn't this be closed as it has been fixed by #916 ?

**Comment by jonathandturner** [comment]
Yup, I'm assuming this is fixed now
